packages/edit-site/src/components/dataviews/view-actions.js

```diff
@@ -139,7 +139,8 @@ function PageSizeMenu( { view, onChangeView } ) {
 
 function FieldsVisibilityMenu( { view, onChangeView, fields } ) {
 	const hidableFields = fields.filter(
-		( field ) => field.enableHiding !== false
+		( field ) =>
+			field.enableHiding !== false && field.id !== view.layout.mediaField
```

Comment:
As a follow-up PR, I'd like to investigate _what if_ we removed `enableHiding` from our API and favor the following logic: all fields are visible unless they are the `view.layout.primaryField` or `view.layout.mediaField`.

The `primaryField` config is only working on the `grid` layout type, but there is no reason it cannot be expanded to all layouts. For example, the `side-by-side` could use it to determine which field that triggers the "side by side" view (in addition to prevent hiding it).

Reply:
>all fields are visible unless they are the view.layout.primaryField or view.layout.mediaField.

I assume you mean all fields are hideable, right? If yes in the future we could have more fields that we don't want to be hideable, but still not the primary one. For example in a product title(primary), price not hideable.

Reply:
That's good, I'll leave it as it is.

Upon sleeping on it, I think what I don't like about the existing `enableHiding` is that is a view-concern that lives in the Field API. The same as `maxWidth`, etc. That may be fine when you only have a single view (TanStack table), but it's not necessarily good when you may have some – each may need different view tweaks.

Ok, so this PR is ready then :)